src/kubernetes/reflector.rs

```diff
@@ -60,8 +67,21 @@ pub async fn custom_reflector<K, W>(
             result = delay_queue.next(), if !delay_queue.is_empty() => {
                 match result {
                     Some(event) => {
-                        trace!(message = "Processing Deleted event.", ?event);
-                        store.apply_watcher_event(&event.into_inner());
+                        let event = event.into_inner();
+                        match event {
+                            watcher::Event::Deleted(ref obj) => {
+                                let meta_desc = MetaDescribe::from_meta(obj.meta());
+                                if meta_cache.contains(&meta_desc) {
+                                    trace!(message = "Skipping event, object is still used.", ?event)
+                                } else {
+                                    trace!(message = "Processing Deleted event.", ?event);
+                                    store.apply_watcher_event(&event);
+                                    // delete information about object meta from meta cacher
+                                    meta_cache.delete(meta_desc);
```

Comment:
`!meta_cache.contains(&meta_desc)` indicates that it has been deleted.

Reply:
done